apps/rush-lib/src/api/packageManager/PnpmPackageManager.ts

```diff
@@ -32,5 +36,17 @@ export class PnpmPackageManager extends PackageManager {
     } else {
       this._shrinkwrapFilename = RushConstants.pnpmV1ShrinkwrapFilename;
     }
+
+    if (parsedVersion.major <= 2) {
+      // node_modules/.shrinkwrap.yaml
+      this.internalShrinkwrapRelativePath = path.join('node_modules', '.shrinkwrap.yaml');
+    } else if (parsedVersion.major <= 3) {
+      // node_modules/.pnpm-lock.yaml
+      this.internalShrinkwrapRelativePath = path.join('node_modules', '.pnpm-lock.yaml');
+    } else {
+      // node_modules/.pnpm/lock.yaml
+      // See https://github.com/pnpm/pnpm/releases/tag/v4.0.0 for more details.
+      this.internalShrinkwrapRelativePath = path.join('node_modules', '.pnpm', 'lock.yaml');
```

Comment:
🤨 Did the also rename the main shrinkwrap file?  Or is this just for the copy under the node_modules folder?

Reply:
just the copy under node_modules

Reply:
Only the one in node_modules folder.